pulp_container/app/schema_convert.py

```diff
@@ -29,18 +30,29 @@ def __init__(self, tag, accepted_media_types, path):
         self.name = path
 
     def convert(self):
-        """Convert a manifest to schema 1."""
+        """Convert a manifest to schema 1.
+
+        Raises:
+            RuntimeError: If the conversion was not successful.
+
+        Returns:
+            ConversionResult: A converted manifest, corresponding digest, and content type.
+
+        """
         if self.tag.tagged_manifest.media_type == MEDIA_TYPE.MANIFEST_V2:
-            converted_schema = self._convert_schema(self.tag.tagged_manifest)
-            return converted_schema, True, self.tag.tagged_manifest.digest
+            schema_with_signature, digest = self._convert_schema(self.tag.tagged_manifest)
+            return ConversionResult(schema_with_signature, digest, MEDIA_TYPE.MANIFEST_V1_SIGNED)
         elif self.tag.tagged_manifest.media_type == MEDIA_TYPE.MANIFEST_LIST:
             legacy = self._get_legacy_manifest()
             if legacy.media_type in self.accepted_media_types:
                 # return legacy without conversion
-                return legacy, False, legacy.digest
+                legacy_schema = _jsonDumps(_get_manifest_dict(legacy))
+                return ConversionResult(legacy_schema, legacy.digest, legacy.media_type)
```

Comment:
So the boolean wasn't even used before?

Reply:
It was used only in this place: https://github.com/pulp/pulp_container/blob/070238111d9e6b297272f94ac1e8d9bc42c89c1b/pulp_container/app/registry.py#L199-L200

Because of that, when we were trying to get, for example, manifest_v2 from manifest_list_v2, it returned an HTTP 404 error because the conversion was not performed. Still, it was a valid request and it should rather return the extracted am64 manifest.